chain/consensus/filcns/upgrades.go

```diff
@@ -1431,6 +1435,149 @@ func upgradeActorsV8Common(
 	return newRoot, nil
 }
 
+func UpgradeActorsCode(ctx context.Context, sm *stmgr.StateManager, newActorsManifestCid cid.Cid, root cid.Cid) (cid.Cid, error) { // nolint
+	buf := blockstore.NewTieredBstore(sm.ChainStore().StateBlockstore(), blockstore.NewMemorySync())
+	store := store.ActorStore(ctx, buf)
+	adtStore := adt8.WrapStore(ctx, store)
+
+	// Load the state root.
+	var stateRoot types.StateRoot
```

Comment:
You can call `LoadStateTree` on the provided root instead. This will give you a StateTree, from which you can more easily `GetActor`, call `system.Load`, etc.